query.go

```diff
@@ -0,0 +1,66 @@
+package duty
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/flanksource/commons/logger"
+)
+
+// Query runs the given SQL query against the provided db connection.
+// The rows are returned as a map of columnName=>columnValue.
+func Query(ctx context.Context, conn *sql.DB, query string) ([]map[string]any, error) {
+	tx, err := conn.BeginTx(ctx, &sql.TxOptions{ReadOnly: true})
```

Comment:
I don't think we need to start and commit a transaction for a readonly query

Reply:
That's just to ensure that only read queries are run.